src/lxc/utils.c

```diff
@@ -330,17 +330,30 @@ int lxc_wait_for_pid_status(pid_t pid)
 	return status;
 }
 
-#if HAVE_LIBGNUTLS
-#include <gnutls/gnutls.h>
-#include <gnutls/crypto.h>
+#ifdef HAVE_OPENSSL
+#include <openssl/evp.h>
 
-__attribute__((constructor))
-static void gnutls_lxc_init(void)
+static int do_sha1_hash(const char *buf, int buflen, unsigned char *md_value, int *md_len)
 {
-	gnutls_global_init();
+	EVP_MD_CTX *mdctx;
+	const EVP_MD *md;
+
+	md = EVP_get_digestbyname("sha1");
```

Comment:
What about using `EVP_sha1()` directly? Its code is AFAIK literally just `return &sha1_md;` so we'd get rid of the error case (which is the main thing which bothers me in gnutls, btw. - many function can't actually really fail, except gnutls has a global "the world has ended" error state...)

Reply:
Hi,

I was going by https://www.openssl.org/docs/man1.1.0/man3/EVP_get_digestbyname.html  which recommended this route.  I don't mind either way.  We might I suppose one day want to make the algorithm configurable (if we ever start using this function for something more security related).

I don't mind either way.